code/datums/components/storage.dm

```diff
@@ -0,0 +1,745 @@
+#define COLLECT_ONE 0
+#define COLLECT_EVERYTHING 1
+#define COLLECT_SAME 2
+
+#define DROP_NOTHING 0
+#define DROP_AT_PARENT 1
+#define DROP_AT_LOCATION 2
+
+// External storage-related logic:
+// /mob/proc/ClickOn() in /_onclick/click.dm - clicking items in storages
+// /mob/living/Move() in /modules/mob/living/living.dm - hiding storage boxes on mob movement
+
+/datum/component/storage
+	var/list/can_hold								//if this is set, only things in this typecache will fit.

```

Comment:
you need to set dupe_mode and handle component inheritance and transfers

Reply:
why? if it's being added twice whoever did it fucked up.

Reply:
what would it even do?